[24.2] Fix some collection builder help and language choices.

jmchilton opened the pull request:
The list help instructions are just wrong and didn't dispatch on the type of components being rendered and even for the right modality was using old names of buttons. That part should be an uncontroversial bug fix. The other commit is language choices - I consider a lot of the old language both too pedantic and then overtly incorrect in places despite that - I prefer all these language choices here more. Incorrect language here and in training documentation and tools has resulted in collections seeming a lot more complex and a lot less beautiful than they are - and I should have worked harder to review everything as it came in. 

## How to test the changes?
(Select all options that apply)
- [x] This is a refactoring of components with existing test coverage.

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

martenson commented on client/src/components/Collections/PairedListCollectionCreator.vue:
```diff
@@ -947,9 +948,9 @@ function _naiveStartingAndEndingLCS(s1: string, s2: string) {
-                                        "Collections of paired datasets are ordered lists of dataset pairs (often forward and reverse reads). ",
-                                        "These collections can be passed to tools and workflows in order to have analyses done on each member of ",
-                                        "the entire group. This interface allows you to create a collection, choose which datasets are paired, ",
+                                        "Lists of pairs are an ordered list of individual dataset paired together in their own collection (often forward and reverse reads). ",
```
```suggestion
                                        "List of pairs is an ordered list of individual datasets paired together in their own collection (often forward and reverse reads). ",
```

martenson replied:
using singular sounds clearer to me here, but feel free to disregard

jmchilton replied:
I think I could agree if it was "A list of pairs" and below "A list" (instead of "This list"). To me "List of pairs is" feels odd. I'll commit that change - it stills satisfies your point that singular would sound better - and I think agree with that.

jmchilton replied:
Ahh... reading it in the broader context I think we are describing lists generally in that first sentence. What if we kept the plural here but we added a sentence at the beginning like "This interface allows you to build a new Galaxy list of pairs. List of pairs are an ordered....". If that still reads odd to you - I'll try something else - I really appreciate your perspective here and for you taking the time to comment. I want this language to read well and feel universal. The language around collections has repeatedly escaped me in ways I'm uncomfortable with.

martenson replied:
I like the added sentence. It is hard (and important imho) to distinguish where we are describing a concrete interface user is looking at and where we explain broader concepts.

martenson commented on client/src/components/Collections/PairedListCollectionCreator.vue:
```diff
@@ -947,9 +948,9 @@ function _naiveStartingAndEndingLCS(s1: string, s2: string) {
-                                        "These collections can be passed to tools and workflows in order to have analyses done on each member of ",
-                                        "the entire group. This interface allows you to create a collection, choose which datasets are paired, ",
+                                        "Lists of pairs are an ordered list of individual dataset paired together in their own collection (often forward and reverse reads). ",
+                                        "These lists can be passed to tools and workflows in order to have analyses done on each member of ",
```
```suggestion
                                        "This list can be passed to tools and workflows in order to have analyses done on each member of ",
```

martenson commented on client/src/components/Collections/ListCollectionCreator.vue:
```diff
@@ -452,17 +453,28 @@ function renameElement(element: any, name: string) {
                                         "and workflows in order to have analyses done on each member of the entire group. This interface allows ",
-                                        "you to create a collection and re-order the final collection.",
+                                        "you to create and re-order a list of datasets. The datasets in a Galaxy collection have an identifier that is preserved accross ",
+                                        "tool executions and server as a form of sample tracking - setting the name in this form will pick the identifier for that element ",
```
```suggestion
                                        "tool executions and serves as a form of sample tracking - setting the name in this form will pick the identifier for that element ",
```

jmchilton committed:
Update client/src/components/Collections/ListCollectionCreator.vue

Co-authored-by: Martin Cech <emulatorer@gmail.com>

jmchilton committed:
Improve language in collection builders...

jmchilton committed:
Fix list collection creator usage instructions.

This was describing the "selection" mode instructions for the free build from history version - if that makes any sense.

jmchilton committed:
Trying to integrate PR comments.

ahmedhamidawan reviewed (approved):
Excellent changes, thank you!
